runelite-client/src/main/java/net/runelite/client/plugins/kourendlibrary/Bookcase.java

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright (c) 2018 Abex
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ *    list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package net.runelite.client.plugins.kourendlibrary;
+
+import java.util.HashSet;
+import java.util.Set;
+import lombok.Getter;
+import lombok.RequiredArgsConstructor;
+
+@RequiredArgsConstructor
+class Bookcase
+{
+	@Getter
+	private final Point3D location;
+
+	@Getter
+	private final int index;
+
+	@Getter
+	private boolean isBookSet;
+
+	/**
+	 * Book in this bookcase as found by the player.
+	 * Will be correct as long as isBookSet is true, unless the library has reset;
+	 */
+	@Getter
+	private Book book;
+
+	/**
+	 * Books that can be in this slot. Will only be populated if library.state != SolvedState.NO_DATA
+	 */
+	@Getter
+	private Set<Book> possibleBooks = new HashSet<>();
+
+	public void clearBook()
+	{
+		book = null;
+		isBookSet = false;
+	}
+
+	public void setBook(Book book)
+	{
+		this.book = book;
+		this.isBookSet = true;
+	}
+
+	public String getLocationString()
+	{
+		StringBuilder b = new StringBuilder();
+
+		boolean north = location.getY() > 3815;
+		boolean west = location.getX() < 1625;
+		if (north && west)
+		{
+			b.append("Northwest");
+		}
+		else if (north)
+		{
+			b.append("Northeast");
+		}
+		else if (west)
+		{
+			b.append("Southwest");
+		}
+		else
+		{
+			b.append("Center");
+		}
+
+		b.append(" ");
+
+		switch (location.getZ())
+		{
+			case 0:
+				b.append("ground floor");
+				break;
+			case 1:
+				b.append("middle floor");
+				break;
+			case 2:
+				b.append("top floor");
+				break;
+		}
+
+		if (KourendLibraryPlugin.debug)
```

Comment:
This should probably be romoved?

Reply:
I would like to keep it there in case the library changes in the future.